lib/ansible/modules/files/iso_extract.py

```diff
@@ -86,12 +86,16 @@
 
 import os.path
 import shutil
+import sys
 import tempfile
 
 try:  # python 3.3+
     from shlex import quote
 except ImportError:  # older python
     from pipes import quote
+finally:
+    if PY2:
+        sys.exc_clear()  # Avoid false positive traceback in fail_json() on Python 2
```

Comment:
This one looks good except that it should be placed i nthe ImportError handling instead of the finally.  (I'm not sure if it makes a difference for this particular case... I think that python will print the exception with the traceback regardless of whether we call sys.exc_clear() in the finally.  But conceptually, if ```from shlex import quote``` raised something other than ImportError we would want to see the traceback from that so calling sys.exc_clear() in that case doesn't seem correct.